### PR TITLE
New Raspberry Pi Tutorial

### PR DESCRIPTION
**Pull request by TylerADavis**
Thanks for contributing to DLR! By submitting this pull request, you confirm that your contribution is made under the terms of the [Apache 2.0 license](https://github.com/neo-ai/neo-ai-dlr/blob/master/LICENSE).

Please refer to our [guideline](https://github.com/neo-ai/neo-ai-dlr/blob/master/CONTRIBUTING.md) for useful information and tips.

This commit introduces a new Jupyter Notebook-based tutorial that shows how to programmatically interact with the Neo compiler and allows users to see the latency improvements Neo brings on their own Raspberry Pi.

Dependent on #252 .

### CODE REVIEW

**Comment by TylerADavis** [comment]
One note: I currently have the below snippet in the tutorial, where the user must manually provide a unique bucket name. Would it be preferable to automatically generate a bucket name so that the reader doesn't need to make any changes to the code cells in the notebook? For example, naming the bucket `raspi_tutorial_{current_time}` or `raspi_tutorial_{random_string}`.

```python
# First, we need to give our bucket a unique name:

# NOTE: This bucket must be renamed as S3 bucket names must be globally unique across all AWS accounts. As the default bucket name below has been used before, bucket creation will fail.

# TODO: manually rename this bucket
bucket_name = 'dvisnty-bucket-party'
```

File: sagemaker-neo-notebooks/edge/raspberry-pi/neo-rasb3b-object-detection.ipynb [hunk 1]
```diff
@@ -0,0 +1,1157 @@
...
+      "Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/pi/.local/lib/python3.7/site-packages (from google-auth<2,>=1.6.3->tensorboard<2.1.0,>=2.0.0->tensorflow) (0.2.8)\n",
+      "Requirement already satisfied: cachetools<5.0,>=2.0.0 in /home/pi/.local/lib/python3.7/site-packages (from google-auth<2,>=1.6.3->tensorboard<2.1.0,>=2.0.0->tensorflow) (4.0.0)\n",
+      "Requirement already satisfied: rsa<4.1,>=3.1.4 in /home/pi/.local/lib/python3.7/site-packages (from google-auth<2,>=1.6.3->tensorboard<2.1.0,>=2.0.0->tensorflow) (4.0)\n",
+      "Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/lib/python3/dist-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.1.0,>=2.0.0->tensorflow) (1.0.0)\n",
+      "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /home/pi/.local/lib/python3.7/site-packages (from pyasn1-modules>=0.2.1->google-auth<2,>=1.6.3->tensorboard<2.1.0,>=2.0.0->tensorflow) (0.4.8)\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip3 install dlr numpy pillow tensorflow matplotlib"
```
Comment (TylerADavis) [inline sagemaker-neo-notebooks/edge/raspberry-pi/neo-rasb3b-object-detection.ipynb:666]: dlr for ARM is not available using pip from pypi. We should instead pip install from https://neo-ai-dlr-release.s3-us-west-2.amazonaws.com/v1.3.0/pi-armv7l-raspbian4.14.71-glibc2_24-libstdcpp3_4/dlr-1.3.0-py3-none-any.whl

**Comment by TylerADavis** [comment]
The DLR version is currently hardcoded to 1.3.0 due to how DLR installed via pip must explicitly use a URL. This doesn't seem ideal to me.

File: sagemaker-neo-notebooks/edge/raspberry-pi/neo-rasb3b-object-detection.ipynb (hunk 1 above)
Comment (TylerADavis) [reply sagemaker-neo-notebooks/edge/raspberry-pi/neo-rasb3b-object-detection.ipynb]: Resolved, but see https://github.com/neo-ai/neo-ai-dlr/pull/254#issuecomment-715585950

**Comment by TylerADavis** [comment]
Spoke with Trevor, he said that since Neo includes a `libdlr.so` with the compiled model, there should be no compatibility issues, and as such the hardcoded DLR is okay.

**Review by trevor-m — APPROVED** [review]
Thanks!

**Pull request merged**